tests/e2e/creation_test.go

```diff
@@ -325,11 +325,17 @@ func (tc *testContext) testApplicationCreation(component components.ComponentInt
 			log.Printf("waiting for application deployments to be in Ready state.")
 			return false, nil
 		}
+
+		// It's ok not to have deployements for unmanaged component
+		if component.GetManagementState() != operatorv1.Managed {
```

Comment:
should the whole test be:

```
if component.GetManagementState() == Managed  {
 loop till deployment.Status.ReadyReplicas == len(appList.Items)  or fail as timeout
 } 
else  { 
return true, nil 
}
```

Reply:
Probably a bit different since appList.Items is the list of Deployments, and every one of them has replicas. But a question, should we wait for all replicas ready or one is enough per Deployment?

Reply:
forget about this comment ^ 

i mixed up `len(appList.Items) `and `deployment.Status.ReadyReplicas `
talking about reality:
- len(appList.Items):  happens for workbnech, kserve and modelserve
- deployment.Status.ReadyReplicas: is for dashboard kserve and modelserve

before this "DSC should be true till all components ready", we did not need to count if all replicas are ready
but now since the logic is to wait for all replicas, then we have to have longer loop.
otherwise the test case is not testing the function